core/templates/domain/story/StoryContentsObjectFactory.ts

```diff
@@ -102,7 +102,8 @@ export class StoryContents {
     });
   }
 
-  getNodeIndex(nodeId: string): number {
+  // If the node id is null, empty or undefined the function returns -1.
+  getNodeIndex(nodeId: string | null | undefined): number {
```

Comment:
Would it be possible to limit this to at least only one of `null` or `undefined`?

Reply:
A initial nodeId can be null if no node exists so null for that case and 
![image](https://user-images.githubusercontent.com/60519303/126204339-47cf831c-4d0e-404b-a80c-ec53d7fd1a49.png)
`.shift()` can return an undefined value so covering that case also. Also I think it would be better if we will use both null and undefined as we're covering all cases then and return -1 for all invalid cases.

Reply:
Ok, explain in the comment above why nodeId can be these values.

Reply:
done